app/models/concerns/conclusion_reviews/upl_pdf.rb

```diff
@@ -0,0 +1,512 @@
+
+      repeated = findings.not_revoked.where.not repeated_of_id: nil
+      present  = findings.not_revoked.where repeated_of_id: nil
+
+      put_upl_repeated_findings_by_risk_on pdf, repeated
+      put_upl_present_findings_by_risk_on  pdf, present
+    end
+
+    def put_upl_repeated_findings_by_risk_on pdf, findings
+      pdf.move_down (PDF_FONT_SIZE * 0.75).round
+      pdf.add_title I18n.t('conclusion_review.repeated_findings'),
+        (PDF_FONT_SIZE * 1.15).round
+
+      if findings.any?
+        put_upl_findings_sorted_by_risk_on pdf, findings
+      else
+        pdf.move_down PDF_FONT_SIZE
+        pdf.text I18n.t('conclusion_review.repeated_findings_empty'),
+          style: :italic
+      end
+    end
+
+    def put_upl_present_findings_by_risk_on pdf, findings
+      reset_last_displayed_control_objective
+
+      pdf.move_down (PDF_FONT_SIZE * 0.75).round
+      pdf.add_title I18n.t('conclusion_review.present_findings'),
+        (PDF_FONT_SIZE * 1.15).round
+
+      if findings.any?
+        put_upl_findings_sorted_by_risk_on pdf, findings
+      else
+        pdf.move_down PDF_FONT_SIZE
+        pdf.text I18n.t('conclusion_review.present_findings_empty'),
+          style: :italic
+      end
+    end
+
+    def put_upl_findings_sorted_by_risk_on pdf, findings
+      findings.sort_for_review.each do |finding|
+        coi = finding.control_objective_item
+
+        put_upl_control_objective_table_on pdf, coi, coi.process_control
+
+        pdf.move_down PDF_FONT_SIZE
+        pdf.text coi.finding_pdf_data(finding), align: :justify, inline_format: true
+      end
+    end
+
+    def put_upl_control_objective_findings_on pdf, grouped_control_objectives, type, use_finals
+      grouped_control_objectives.each do |process_control, cois|
+        has_findings = has_findings_for_review? cois, type, use_finals
+
+        if has_findings
+          cois.sort.each do |coi|
+            coi_findings = coi_findings_for coi, type, use_finals
+
+            if coi_findings.not_revoked.present?
+              findings = coi_findings.not_revoked.sort_for_review
+
+              put_upl_control_objective_table_on pdf, coi, process_control
+
+              findings.each do |f|
+                pdf.move_down PDF_FONT_SIZE
+                pdf.text coi.finding_pdf_data(f), align: :justify, inline_format: true
+              end
+            end
+          end
+        end
+      end
+    end
+
+    def has_findings_for_review? control_objective_items, type, use_finals
+      control_objective_items.any? do |coi|
+        findings = coi_findings_for coi, type, use_finals
+
+        findings.not_revoked.present?
+      end
+    end
+
+    def upl_control_objective_column_data_for control_objective_item, process_control
+      caption = ControlObjective.model_name.human
+      data    = []
+
+      data << upl_finding_column_headers(process_control)
+      data << ["<b>#{caption}:</b> #{control_objective_item.to_s}\n"]
+
+      data
```

Comment:
```suggestion

      [
        upl_finding_column_headers(process_control),
        ["<b>#{caption}:</b> #{control_objective_item.to_s}\n"]
      ]
```
Si no me equivoco es lo mismo.